src/charm_state.py

```diff
@@ -224,6 +227,30 @@ def to_yes_or_no(cls, value: str) -> str:
             return "yes"
         return "no"
 
+    @validator("publish_rooms_allowlist")
+    @classmethod
+    def publish_rooms_allowlist_to_list(cls, value: str) -> typing.List[str]:
+        """Convert the publish_rooms_allowlist field to list.
+
+        Args:
+            value: the input value.
+
+        Returns:
+            The string converted to list.
+
+        Raises:
+            ValidationError: if user_id is not as expected.
+        """
+        if value is None:
+            return []
+        allow_list = [user_id.strip() for user_id in value.split(",")]
+        for user_id in allow_list:
+            if not re.match(r"@\w+:\w+\.\w+", user_id):
```

Comment:
I'm wondering if this regex could be a constant and have a comment explaining what it does help - although this seems simple enough :)
FYI, I've had issues with `w+` pattern with it not allowing dashes, just checking in if you want to allow dashes in user ids :D
Thanks!

Reply:
Excellent catch, this helped a lot. After your review, I checked on the spec that actually "-" (and other symbols) are allowed.
Also, instead of creating a constant, I applied the validator for the two configuration items that use it.